### DISCUSSION TITLE
tensor rotation

### DISCUSSION THREAD

**Discussion by jessecarterMOOSE** [idaholab/moose#16252] [Q&A Modules: Solid mechanics] [answered]
Hi all. I wanted to talk about tensor rotations (again) as it pertains to mechanics simulations, but without a discussion on those pesky Euler angles. It seems all you need is a rotation matrix and some clever manipulation to rotate the tensor (easy in python). Now I already know the basis vectors for how my crystal phase is embedded in my matrix phase, so there are really only two rotation matrices possible - the basis vectors are either rows or columns. One takes a vector direction in the matrix (sample) phase and gives me a direction in the crystal phase, and vice versa. After some googling, which one you want depends on the application. So for the case of straining a sample and getting the proper stresses inside of the crystal phase, which way do I go?

**Comment by aeslaughter** [comment, marked as answer]
Where you able to find an answer to this?

**Reply by jessecarterMOOSE** [reply]
I tried to figure it out myself and ended up submitting #16341. If anybody has any comments I would ask them to post there.
